packages/integration-tests/src/EntityManager.test.ts

```diff
@@ -1023,8 +1023,105 @@ describe("EntityManager", () => {
     // Then the deleted entity was returned from the flush
     expect(result).toEqual(a1);
   });
+
+  it("can clone entities", async () => {
+    const em = newEntityManager();
+
+    // Given an entity
+    const a1 = new Author(em, { firstName: "a1" });
+    await em.flush();
+
+    // When we clone that entity
+    const a2 = await em.clone(a1);
+
+    // Then we expect the cloned entity to have the same properties as the original
```

Comment:
This is checking that a primitive was cloned, could we create a publisher, pass that to the `Author` cstr on line 1031, and then after cloning (with the existing no-populate-hint clone call), add an assertion that `a2.publisher.id === p1.id`? And maybe that `numberOf([Author, Publisher])` is now `[2, 1]`?

Fwiw I think `clone` is already handling this correctly, just thinking I'd like to have it covered by a test, and I didn't see an existing test for this (although cloning the author -> book -> image probably gets really close/probably does...)

Reply:
That makes sense to add coverage for, I'll add a publisher in